src/main/drivers/pwm_output.c

```diff
@@ -193,13 +193,17 @@ void pwmEnableMotors(void)
 
 void pwmCompleteOneshotMotorUpdate(uint8_t motorCount)
 {
-    TIM_TypeDef *lastTimerPtr = NULL;
-
     for (int index = 0; index < motorCount; index++) {
-        // Force the timer to overflow if it's the first motor to output, or if we change timers
-        if (motors[index]->tim != lastTimerPtr) {
-            lastTimerPtr = motors[index]->tim;
-            timerForceOverflow(motors[index]->tim);
+        bool overflowed = false;
+        // If we have not already overflowed this timer
+        for (int j = 0; j < index; j++) {
```

Comment:
it could be a constant time implementation... map the timers to an indexed array and lookup with O(1) maybe, flag when the timer is overflowed and cleared after the whole loop maybe? but that can pry wait for a future improvement